firmware/usb.c

```diff
@@ -426,9 +427,10 @@ char usbTiny(char set)
 	return old;
 }
 
-void usbDelay(int cycles)
+void usbSleep(uint32_t millis)
 {
-	while (cycles--) {
+	uint32_t end = system_millis + millis;
+	while (end > system_millis) {
```

Comment:
It's probably not that critical, but the timer runs over every 4 million seconds, i.e. roughly every 50 days (if someone keeps the Trezor plugged in that long).

To avoid this, use `(int32_t) (end - system_millis) > 0` or `(system_millis - start) < millis`